app/scripts/components/add_project.component.js

```diff
@@ -53,8 +53,10 @@
           projectaanvraagApiService.addProject(ctrl.formData).then(function() {
             // Show success message and redirect to the dashboard
             Messages.addMessage('success', 'Je project is aangemaakt. Je vindt het hieronder in de lijst terug.');
-            ctrl.saving = false;
-            $state.go('authenticated.dashboard');
+              $timeout(function(){
+                ctrl.saving = false;
+                $state.go('authenticated.dashboard');
+              }, 500);
```

Comment:
Why would there be a delay in the creation of the project? If the request has been executed and a response is received it should be created no?
If not, is there another way to verify it's been created? This timeout is arbitrary and could still be too short, while increasing it would make the experience slower and slower.

Reply:
@bertramakers, I totally agree with you. Adding the timeout seems dirty, however I don't see another solution, since the creation of the insightly id is being handled by rabbitmq. See jira ticket for more context.
If you have better idea, suggestions are welcome.

Reply:
Ah right we're waiting for the insightly id. Could we improve this by adding some sort of polling mechanism on the dashboard to attempt to fetch the insightly id every x seconds if it wasn't included in the initial data (because it wasn't created yet) and temporarily show a loading indicator next to that project? If it's not feasible now, maybe create a follow-up ticket to pick this up later. Because especially with RabbitMQ in-between we cannot guarantee that it will be created after 500ms.

Reply:
@bertramakers good idea! I've discussed this with Corneel and we're going to pick it up in  a seperate ticket.